packages/salesforcedx-vscode-soql/src/queryDataView/dataProviders/csvDataProvider.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { JsonMap } from '@salesforce/ts-types';
+import * as Papa from 'papaparse';
+import { DATA_CSV_EXT } from '../../constants';
+import { DataProvider } from './iDataProvider';
+
+export class CsvDataProvider implements DataProvider {
+  public readonly fileExtension = DATA_CSV_EXT;
+  constructor(public readonly documentName: string) {}
+
+  public getFileContent(data: JsonMap[]): string {
+    const queryRecordsCsv = Papa.unparse(data, {
+      header: true,
+      delimiter: ','
+    });
```

Comment:
[PapaParse](https://www.papaparse.com/docs#json-to-csv)! (love the name)

If an error is thrown due to bad JSON format it will be caught by the caller.
This will produce something like:
```
Column 1,Column 2
1-1,1-2
2-1,2-2
3-1,3-2
4,5
```